### ISSUE TITLE
Enable/Disable the "Print Only" for the end of post Footnote list

### ISSUE THREAD

**Issue by chrisspiegl** [goblindegook/littlefoot#380]
I would like to see an additional option in the setup config to not hide the footnote list at the end of the post. The idea is, if someone wants to read all the footnotes at the end, they can.

I have in the meantime found a workaround to force things visible again — which isn't all that easy since they are forced invisible by the `display: none !important`.

My workaround for people that are curious, it keeps the numbers in the text invisible, but adds the list of footnotes including their index back in at the bottom:

```css
.footnotes.footnote-print-only,
.footnotes-list.footnote-print-only {
  display: block !important;
}
.footnote-item.footnote-print-only {
  display: list-item !important;
}
```

**Comment by goblindegook** [comment]
Thanks for the suggestion. I would need a really compelling reason to introduce a new setting, as configuration tends to add to complexity and make the project harder for me to test and maintain.

An option to show the original footnotes seems simple but opens some potential for complication, bugs and eventually frustration all around. The more I thought about it, the more questions it raised. For example, would it mean keeping all footnote elements, even if it meant displaying the original callouts alongside the buttons?

<img width="95" alt="image" src="https://user-images.githubusercontent.com/285497/85881899-4b27b800-b7d6-11ea-8603-7e265326c6ec.png">

This looks a bit weird, right? Would users expect or appreciate this?

Or perhaps it would keep only the footnotes at the end. But considering the default button template uses ellipses instead of footnote numbers, yet the list still displays numerical references, wouldn't that be weird as well? Should there be different template defaults depending on whether a user keeps the original footnotes? Or should I change the default for everyone, potentially giving them an unpleasant surprise on upgrade?

It's hard to make these decisions with limited information, and chasing down the answers requires time I don't currently have.

Fortunately, a developer is already able to override the library's behaviour and fine tune it to their particular liking, either by creating CSS definitions like you did, or by removing the `footnote-print-only` class from the elements immediately after instantiation, for example:

```js
[...document.querySelectorAll('.footnote-print-only')].forEach(e => e.classList.remove('footnote-print-only'))
```

I'll keep your feature request in mind, and it would be good to know if there is much call for it, but it's just not something I plan to add to the project any time soon.

**Comment by chrisspiegl** [comment]
Hello @goblindegook, thank you for taking the time to think about this.

I understand the challenge that you are writing about in terms of adding more variables that need testing and more.

I personally saw the type of setup I admire and want to work toward on the blog by Marco Arment at [marco.org](https://marco.org). He has it set up to show the number in the footnote reference button and then at the end of the blog post, he also shows the footnote list.

I already made that work with some CSS and of course the use of the custom button template.

With the help of your comment, especially regarding the ability to remove the `.footnote-print-only`, I was able to even simplify my "workaround" a bit more to look like this:

```js
    littlefoot.default({
      buttonTemplate: `<sup
                        aria-expanded="false"
                        aria-label="Footnote <% number %>"
                        class="footnote-ref">
                        <a
                        href="#<% reference %>"
                        class="footnote-ref"
                        id="<% reference %>"
                        data-footnote-identifier="<% reference %>"
                        data-footnote-style="default"
                        alt="See Footnote <% number %>"
                        rel="footnote"
                        data-footnote-content="<% content %>">
                            <% number %>
                    </a></sup>`
    })
    document.querySelectorAll('.footnotes.footnote-print-only, .footnotes .footnote-print-only').forEach(e => e.classList.remove('footnote-print-only'))
    document.querySelectorAll('.footnote-backref').forEach((el) => { el.href = el.href.replace('#', '#lf-') })
```
This code basically makes sure that:

-  the footnote reference looks and codes the same as the original markdown rendered footnote reference
- it removes the `.footnote-print-only` class from all the elements at the end of the blog post, but keeps the little numbers invisible inside the post since those are already taken care of by the number inside the template
- and, last but not least, it also makes sure that the footnotes at the bottom of the blog post link the newly inserted Littlefoot references with are prefixed by `lf-`.

If anybody has any suggestions/improvements for this, I am very interested to hear them.

A question for example that came up for me was whether or not I "need" the `[...document…].forEach` wrapper, because it does work without it and I'd rather use a more universal code (Safari complains about those `[...var]` spreads.

Thanks again for all the help, and I hope this helps someone else as well.

**Comment by goblindegook** [comment]
> A question for example that came up for me was whether or not I "need" the `[...document…].forEach` wrapper, because it does work without it and I'd rather use a more universal code (Safari complains about those `[...var]` spreads.

Ah, no, mine was just a quick example, and it doesn't even use universally compatible syntax. Mind you, not all browsers support `NodeList#forEach` either, you might want to try something like [the examples on MDN](https://developer.mozilla.org/en-US/docs/Web/API/NodeList).

**Comment by chrisspiegl** [comment]
Thank you for that followup. I have adjusted my last two lines like so:

```js
Array.prototype.forEach.call(document.querySelectorAll('.footnotes.footnote-print-only, .footnotes .footnote-print-only'), e => e.classList.remove('footnote-print-only'))
Array.prototype.forEach.call(document.querySelectorAll('.footnote-backref'), (el) => { el.href = el.href.replace('#', '#lf-') })
```

It still works and looks like based on the MDN resource is more reliable.

Regarding older browsers, honestly, it doesn't matter too much for me. My current estimate is that if it does not work, then it's not a biggy because it simply falls back to: footnotes are not visible at the end of the post.

Closing this since it's done from my side. If it ever gets added as an extension, maybe this can be discussed in a new issue.
